### DISCUSSION TITLE
Is GraphQL server implementation of GitHub API v4 open source?

### DISCUSSION THREAD

**Discussion by kedarmhaswade** [community/community#24457] [Apps, API and Webhooks]
<p>Can someone point me to code if it is open source? If it is not open source, is there some documentation to look at?</p>
<p>Regards,</p>
<p>Kedar</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/kedarmhaswade">@kedarmhaswade</a>,</p>
<p>Thanks for being here! Here is a link to <a href="https://developer.github.com/v4/" rel="nofollow noopener">v4’s documentation</a>. I hope this helps!</p>

**Comment by kedarmhaswade** [comment]
<p>Thanks for your response and the link.</p>
<p>I was looking for some more details on the <em>implementation</em> of the GitHub GraphQL (v4) API. How is it implemented? Is it based on Apollo?</p>

**Comment by kedarmhaswade** [comment, marked as answer]
<p>Found it at <a href="https://githubengineering.com/the-github-graphql-api/" title="The GH GraphQL API" rel="nofollow noopener">the GitHub Engineering Blog Post</a>. GitHub seems to use <a href="https://graphql-ruby.org/" rel="nofollow noopener">https://graphql-ruby.org/</a> (<a href="https://github.com/rmosolgo/graphql-ruby" rel="nofollow noopener">mosolgo/graphql-ruby</a> gem).</p>
